packages/core/src/shipping/strategies/braintree/braintree-accelerated-checkout-shipping-strategy.ts

```diff
@@ -0,0 +1,221 @@
+        const { clientToken, initializationData } = paymentMethod;
+
+        if (!clientToken || !initializationData) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        this._braintreeIntegrationService.initialize(clientToken, initializationData);
+
+        const braintreeConnect = await this._braintreeIntegrationService.getBraintreeConnect(
+            cart?.id,
+        );
+
+        const customerEmail = customer?.email || billingAddress?.email;
+
+        if (!customerEmail) {
+            return;
+        }
+
+        const { lookupCustomerByEmail, triggerAuthenticationFlow } = braintreeConnect.identity;
+
+        const { customerContextId } = await lookupCustomerByEmail(customerEmail);
+
+        if (!customerContextId) {
+            return;
+        }
+
+        const { authenticationState, profileData } = await triggerAuthenticationFlow(
+            customerContextId,
+        );
+
+        const addresses = this._mapPayPalToBcAddress(profileData.addresses, countries) || [];
+        const instruments = this._mapPayPalToBcInstrument(methodId, profileData.cards) || [];
+
+        await this._store.dispatch(
+            this._paymentProviderCustomerActionCreator.updatePaymentProviderCustomer({
+                authenticationState,
+                addresses,
+                instruments,
+            }),
+        );
+
+        if (addresses.length > 0) {
+            await this._store.dispatch(
+                this._billingAddressActionCreator.updateAddress({
+                    ...addresses[0],
+                    id: String(addresses[0].id),
+                }),
+            );
+        }
+
+        if (addresses.length > 0 && cart.lineItems.physicalItems.length > 0) {
+            await this._store.dispatch(this._consignmentActionCreator.updateAddress(addresses[0]));
+        }
+    }
+
+    private _getCountryNameByCountryCode(countryCode: string, countries: Country[]): string {
+        const matchedCountry = countries.find((country) => country.code === countryCode);
+
+        return matchedCountry?.name || '';
+    }
+
+    private _mapPayPalToBcAddress(
+        addresses: BraintreeConnectAddress[],
+        countries: Country[],
+    ): CustomerAddress[] | undefined {
+        return addresses.map((address) => ({
+            id: Number(address.id),
+            type: 'paypal-address',
+            firstName: address.firstName || '',
+            lastName: address.lastName || '',
+            company: address.company || '',
+            address1: address.streetAddress,
+            address2: address.extendedAddress || '',
+            city: address.locality,
+            stateOrProvince: address.region,
+            stateOrProvinceCode: address.region,
+            country: this._getCountryNameByCountryCode(address.countryCodeAlpha2, countries),
+            countryCode: address.countryCodeAlpha2,
+            postalCode: address.postalCode,
+            phone: '',
```

Comment:
What happens in the case for stores that have phone number as a required property?

Reply:
In this case will be shown form where user will add a phone number, because we don't change the basic checkout flaw, we only complement it
https://github.com/bigcommerce/checkout-js/blob/master/packages/core/src/app/shipping/ShippingAddressForm.tsx#L80